sklearn/linear_model/_least_angle.py

```diff
@@ -966,6 +968,19 @@ class Lars(MultiOutputMixin, RegressorMixin, LinearModel):
     [ 0. -1.11...]
     """
 
+    _parameter_constraints = {
+        "fit_intercept": ["boolean"],
+        "verbose": ["verbose"],
+        "normalize": ["boolean", Hidden(StrOptions({"deprecated"}))],
+        "precompute": ["boolean", StrOptions({"auto"}), "array-like", Hidden(None)],
```

Comment:
I checked and I think that the documentation is wrong here. We cannot accept array-like but only `ndarray`. I think this is fine since this is an initialization. We should only change the documentation. Here is the minimal reproducer:


```python
from sklearn.datasets import make_regression
from sklearn.linear_model import Lars

X, y = make_regression()
gram = (X.T @ X).tolist()
Lars(precompute=gram).fit(X, y)
```

It will fail with:

```pytb
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-9-ea9c494be72a> in <module>
----> 1 lars = Lars(precompute=gram).fit(X, y)

~/Documents/packages/scikit-learn/sklearn/linear_model/_least_angle.py in fit(self, X, y, Xy)
   1143             y = y + noise
   1144 
-> 1145         self._fit(
   1146             X,
   1147             y,

~/Documents/packages/scikit-learn/sklearn/linear_model/_least_angle.py in _fit(self, X, y, max_iter, alpha, fit_path, normalize, Xy)
   1045             for k in range(n_targets):
   1046                 this_Xy = None if Xy is None else Xy[:, k]
-> 1047                 alphas, active, coef_path, n_iter_ = lars_path(
   1048                     X,
   1049                     y[:, k],

~/Documents/packages/scikit-learn/sklearn/linear_model/_least_angle.py in lars_path(X, y, Xy, Gram, max_iter, alpha_min, method, copy_X, eps, copy_Gram, verbose, return_path, return_n_iter, positive)
    169             "Use lars_path_gram to avoid passing X and y."
    170         )
--> 171     return _lars_path_solver(
    172         X=X,
    173         y=y,

~/Documents/packages/scikit-learn/sklearn/linear_model/_least_angle.py in _lars_path_solver(X, y, Xy, Gram, n_samples, max_iter, alpha_min, method, copy_X, eps, copy_Gram, verbose, return_path, return_n_iter, positive)
    488     else:
    489         n_features = Cov.shape[0]
--> 490         if Gram.shape != (n_features, n_features):
    491             raise ValueError("The shapes of the inputs Gram and Xy do not match.")
    492 

AttributeError: 'list' object has no attribute 'shape'
```

Because we expect a NumPy array.

Reply:
```suggestion
        "precompute": ["boolean", StrOptions({"auto"}), np.ndarray, Hidden(None)],
```

Reply:
This should apply and below